Lib/tkinter/__init__.py

```diff
@@ -1527,10 +1527,21 @@ def bind(self, sequence=None, func=None, add=None):
         return self._bind(('bind', self._w), sequence, func, add)
 
     def unbind(self, sequence, funcid=None):
-        """Unbind for this widget for event SEQUENCE  the
-        function identified with FUNCID."""
-        self.tk.call('bind', self._w, sequence, '')
-        if funcid:
+        """Unbind for this widget the event SEQUENCE.
+
+        If FUNCID is given, only unbind the function identified with FUNCID
+        and also delete that command.
+        """
+        if funcid is None:
+            self.tk.call('bind', self._w, sequence, '')
+        else:
+            lines = self.tk.call('bind', self._w, sequence).split('\n')
```

Comment:
Without seeing what `lines` looks like for various scenarios, I cannot verify the code below.  Hence would have to trust test.  Hence would like testing of what seems like a harder scenario.

Reply:
Usually it looks like:
```
if {"[140011831004144test1 %# %b %f %h %k %s %t %w %x %y %A %E %K %N %W %T %X %Y %D]" == "break"} break

if {"[140011830917168test2 %# %b %f %h %k %s %t %w %x %y %A %E %K %N %W %T %X %Y %D]" == "break"} break

```
if it only contains bindings added by `bind()`. It can also contain arbitrary Tcl commands, including multiline commands, if it is standard binding or `bind()` was called with a string instead of callable (AFAIK IDLE does such hacking). It does not matter, because it is unlikely that somebody add commands in the above form, using generated `funcid`.

Reply:
I checked -- IDLE uses this in `fix_x11_paste()`.

Reply:
It's useful to manually register a string (to reuse functions multiple times or to directly call a Tk method), but that would entirely reasonably preclude using `unbind()` with that function. Should there be a test or something to verify it leaves other lines intact?

Reply:
Right, you wrote that.  This comment only serves to explain what I did not review in detail and why I requested the test change.